src/confd/src/ietf-keystore.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include <sys/stat.h>
+#include <srx/common.h>
+#include <srx/lyx.h>
+
+#include <srx/srx_val.h>
+
+#include "base64.h"
+#include "core.h"
+
+#define XPATH_KEYSTORE_  "/ietf-keystore:keystore/asymmetric-keys"
+
+static size_t filesz(const char *fn)
+{
+	struct stat st;
+
+	if (stat(fn, &st))
+		st.st_size = BUFSIZ;
+	return st.st_size;
+}
+
+/* sanity check, must exist and be of non-zero size */
+static bool fileok(const char *fn)
+{
+	if (!fexist(fn))
+		return false;
+
+	if (filesz(fn) < 42)
+		return false;
+
+	return true;
+}
+
+static char *filerd(const char *fn, size_t len)
```

Comment:
```suggestion
static char *key_from_pem(const char *fn, size_t len)
```
I had to read this many times before I was able to parse "file read" (I think). Maybe something a bit more descriptive?

Reply:
Thanks, added comments to all helper functions.